tests/test_oneway_anova.py

```diff
@@ -36,28 +37,38 @@ def test_v_stats(self):
         m3 = [3 for _ in range(n_features)]
         fd = FDataGrid([m1, m2, m3], sample_points=t)
         self.assertEqual(v_sample_stat(fd, weights), 7.0)
-        self.assertEqual(v_sample_stat(fd, weights, p=1), 5.0)
+        self.assertAlmostEqual(v_sample_stat(fd.to_basis(Fourier(n_basis=5)),
+                                             weights), 7.0)
         res = (1 - 2 * np.sqrt(1 / 2)) ** 2 + (1 - 3 * np.sqrt(1 / 3)) ** 2 \
                                             + (2 - 3 * np.sqrt(2 / 3)) ** 2
         self.assertAlmostEqual(v_asymptotic_stat(fd, weights), res)
-        res = abs(1 - 2 * np.sqrt(1 / 2)) + abs(1 - 3 * np.sqrt(1 / 3))\
-                                          + abs(2 - 3 * np.sqrt(2 / 3))
-        self.assertAlmostEqual(v_asymptotic_stat(fd, weights, p=1), res)
+        self.assertAlmostEqual(v_asymptotic_stat(fd.to_basis(Fourier(
+            n_basis=5)), weights), res)
 
     def test_asymptotic_behaviour(self):
```

Comment:
I hope this is fast enough. Is it?

Reply:
In my computer the test is executed in 18 seconds, I know it may not be faster enough, but maybe we can discuss another type of test in the meeting of tomorrow, as a faster test will not be able to check the asymptotic behaviour of the procedure with a reasonable error range.